src/main/java/de/unijena/cheminf/fragment/fingerprint/performanceTest/PerformanceTest.java

```diff
@@ -0,0 +1,419 @@
+            }
+            // results files
+            File tmpResultsLogFile = new File(this.workingPath + "/Results/" + PerformanceTest.RESULTS_FILE_NAME + tmpProcessingTime + ".txt");
+            FileWriter tmpResultsLogFileWriter = new FileWriter(tmpResultsLogFile, true);
+            this.resultsPrintWriter = new PrintWriter(tmpResultsLogFileWriter);
+            this.resultsPrintWriter.println("#########################################################################");
+            this.resultsPrintWriter.println();
+            this.resultsPrintWriter.println("Processing Time: " + tmpProcessingTime);
+            this.resultsPrintWriter.println();
+            this.resultsPrintWriter.println("Application initialized. Loading  files named " + anArgs[0] + " and "+ anArgs[1] + ".");
+            this.resultsPrintWriter.println();
+            // bit fingerprints process file
+            File tmpBitFingerprintsResultFile = new File(this.workingPath + "/Results/" + PerformanceTest.CSV_BIT_FINGERPRINT_PROCESS_RESULT_FILE_NAME + tmpProcessingTime + ".csv");
+            FileWriter tmpBitFingerprintsResultWriter = new FileWriter(tmpBitFingerprintsResultFile, false);
+            this.bitPrintWriter = new PrintWriter(tmpBitFingerprintsResultWriter);
+            // count fingerprints process file
+            File tmpCountFingerprintsResultFile = new File(this.workingPath + "/Results/" + PerformanceTest.CSV_COUNT_FINGERPRINT_PROCESS_RESULT_FILE_NAME + tmpProcessingTime + ".csv");
+            FileWriter tmpCountFingerprintResultWriter = new FileWriter(tmpCountFingerprintsResultFile, false);
+            this.countPrintWriter = new PrintWriter(tmpCountFingerprintResultWriter);
+            // bit array file
+            File tmpBitArrayFingerprintResultFile = new File(this.workingPath + "/Results/" + PerformanceTest.BIT_ARRAY_RESULT_FILE_NAME + tmpProcessingTime+ ".txt");
+            FileWriter tmpBitArrayFingerprintResultWriter = new FileWriter(tmpBitArrayFingerprintResultFile, false);
+            this.bitArrayPrintWriter = new PrintWriter(tmpBitArrayFingerprintResultWriter);
+            // read in CSV files that contains fragments
+            try {
+                this.importDataFromTextFile();
+            } catch (IOException anException) {
+                this.exceptionsPrintWriter.println("Fragment load ERROR. Unsuitable (structure) CSV files were tried to be read in.");
+                this.exceptionsPrintWriter.flush();
+                this.appendToLogfile(anException);
+                throw new Exception("Fragment load ERROR. Unsuitable (structure) CSV files were tried to be read in. ");
+            }
+            // generate bit and count fingerprints
+            this.generateFingerprints(Integer.parseInt(anArgs[3]),1,2); // TODO delete arguments
+            this.resultsPrintWriter.flush();
+            this.bitPrintWriter.println();
```

Comment:
Please no empty lines at the end of files.

Reply:
Why not?

Reply:
You as the arch enemy of empty lines are asking why? :D

Reply:
Empty lines in the middle of code are not the same as an empty line at the end of a text file. There is a reason for the empty line or to be more precisely the \n at the end. Iirc (old) unix and c(++) programs accepts text files as text files only if they are not empty and ends with a new line. If not they are no text files. The "No new line at end of file" i about not removing the newline that should be there.

Reply:
Ok, so files should have empty lines at the end in general? But shouldn't the programs be able to distinguish between "new line character at the end of the last line of the file" and "empty line at the end of the file" internally? Maybe we discuss this tomorrow in our meeting.